### PR TITLE
Bug 1882107 - Add ability to reorder tabs in tab strip with drag and drop

### PR DESCRIPTION
**Pull request by rahulsainani**
### How
– Make existing ReorderedList support horizontal lists aka LazyRow. This required changes in `DragItemContainer` and drag detect Modifier.
– Use that functionality in TabStrip

### Preview

https://github.com/mozilla-mobile/firefox-android/assets/38040960/0d77bc78-588d-48ee-a288-705bc75d91e4

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1882107

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt [hunk 1]
```diff
@@ -112,6 +120,7 @@ private fun TabStripContent(
     onAddTabClick: () -> Unit,
     onCloseTabClick: (id: String) -> Unit,
     onSelectedTabClick: (id: String) -> Unit,
+    onMove: (String, String, Boolean) -> Unit,
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabStrip.kt:123]: nit: might also be useful to provide names to the params here too

**Review by t-p-white — APPROVED** [review]
Great addition 👌 lgtm

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt [hunk 2]
```diff
@@ -266,17 +276,27 @@ private fun LazyListState.findItem(offset: Float) =
  * @param reorderState List reordering state used for dragging callbacks.
  * @param shouldLongPressToDrag Whether or not an item should be long pressed to start the dragging gesture.
  */
-fun Modifier.detectVerticalPressAndDrag(
+fun Modifier.detectListPressAndDrag(
     listState: LazyListState,
     reorderState: ListReorderState,
     shouldLongPressToDrag: Boolean,
-): Modifier = pointerInput(listState, shouldLongPressToDrag) {
+): Modifier = this then pointerInput(listState, shouldLongPressToDrag) {
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt:283]: the inline `then` is pretty neat. TIL

**Review by MozillaNoah — APPROVED** [review]
LGTM! It works in the previews, too!!

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt (hunk 2 above)
Comment (t-p-white) [reply fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt:283]: @MozillaNoah ➕ I initially thought it was a typo 🙃

**Pull request merged**